src/peerconnection.cpp

```diff
@@ -694,11 +694,23 @@ void PeerConnection::openTracks() {
 	if (auto transport = std::atomic_load(&mDtlsTransport)) {
 		auto srtpTransport = std::reinterpret_pointer_cast<DtlsSrtpTransport>(transport);
 		std::shared_lock lock(mTracksMutex); // read-only
-		for (auto it = mTracks.begin(); it != mTracks.end(); ++it)
-			if (auto track = it->second.lock()) {
-			    if (!track->isOpen())
+//		for (auto it = mTracks.begin(); it != mTracks.end(); ++it)
+        for (unsigned int i = 0; i < mTrackLines.size(); i++) {
+            if (auto track = mTrackLines[i].lock()) {
+                if (!track->isOpen()) {
+//                    if (track->description().direction() == rtc::Description::Direction::RecvOnly || track->description().direction() == rtc::Description::Direction::SendRecv)
+//                        srtpTransport->addInboundSSRC(0);
+//                    if (track->description().direction() == rtc::Description::Direction::SendOnly || track->description().direction() == rtc::Description::Direction::SendRecv)
+
+                    for (auto ssrc : track->description().getSSRCs())
+                        srtpTransport->addSSRC(ssrc);
+                    for (auto ssrc : std::get<rtc::Description::Media *>(remoteDescription()->media(i))->getSSRCs())
```

Comment:
We should figure out a way to make these two statements a lot less messy. There needs to be a function called `activeDescription` that returns a mixture of both descriptions.

Reply:
The logic looks weird to me. Why is the second for loop in the track loop, since it doesn't even depend on the track?

Reply:
It is indirectly related to the track. In essence it would have the same functionality as `track->remoteDescription()->getSSRCs()` if remoteDescription was a function inside of Track.

Reply:
Yes however currently it sets up all remote tracks and only requires the local description, so why not doing it only once after the track loop?

Reply:
Hmm, where would you recommend I move it too then?